File: files/CSEC 14.07.R
library(tidyverse)
library(readxl)

path = "files/Excel Challenge 14th July.xlsx"
input = read_xlsx(path, range = "B2:D8")
test  = read_xlsx(path, range = "F2:L8")

result = input %>%
  separate_rows(c(Customers, Orders), sep = "; ") %>%
  pivot_wider(names_from = Customers, values_from = Orders) %>%
  mutate(across(-c(1), ~as.numeric(.)))

all.equal(result, test)
#> [1] TRUE

File: files/2024-07-14/CSEC 14.07.py
import pandas as pd

path = "files/Excel Challenge 14th July.xlsx"
input = pd.read_excel(path, usecols="B:D", skiprows=1, nrows = 6)
test = pd.read_excel(path, usecols="F:L", skiprows=1, nrows = 6)
test.columns = test.columns.str.replace(".1", "")

result = input.assign(Customers=input.Customers.str.split("; "),
                     Orders=input.Orders.str.split("; ")) \
    .explode(["Customers", "Orders"]) \
    .pivot(index="Date", columns="Customers", values="Orders") \
    .reset_index()

result[result.columns[1:]] = result[result.columns[1:]].apply(pd.to_numeric).fillna(" ")
result.columns.name = None
test = test.fillna(" ")

print(result.equals(test)) # True